ddev/ddev
# build: push gitpod and codespaces images/packages during release

## The Issue

We've had the gitpod image and codespaces package pushes as separate tasks for some time, and they weren't always done. 

## How This PR Solves The Issue

* Push gitpod image after release build
* Push codespaces/vscode package after release build
* If building on ddev-test, push images to ddevhq Docker org so things are safer (the push happens to the variable DOCKER_ORG)
* Rename containers_shared.mak to containers_shared.mk so that it's recognized as a Makefile and tabs get used
* Adds a `latest` tag for Docker images for stable releases. We don't need to use these, and probably shouldn't, but this will remove maintenance of ddev-gitpod-launcher, and may help with how things work for drupalpod eventually. There is a place for `latest`

## Notes

I was overly ambitious here and tried to create a multi-architecture image of the upstream gitpod base image. That attempt was removed in https://github.com/ddev/ddev/pull/6613/commits/608a00d92476fa706d2ce5a9c38268c374f4aaf0 because it didn't seem to work quite right. I was trying to recreate the gitpod-sourced image with an alternate build technique, not using their `dazzle` technique. It didn't seem to me that it was working completely correctly, and I think my proposed use of the arm64 image was probably not as useful as expected.

## Manual Testing Instructions

* Build a release with normal release process on ddev-test. See https://github.com/ddev-test/ddev/releases/tag/v1.23.5-aaa-testgitpod.21 and https://github.com/ddev-test/ddev/actions/runs/11346993948
* Manual push of ddev-gitpod-base done in https://github.com/rfay/ddev/actions/runs/11346980563
